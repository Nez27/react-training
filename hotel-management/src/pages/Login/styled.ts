import styled from 'styled-components';

// Styled
import CommonInput from '@src/commons/styles/CommonInput';

const LoginLayout = styled.main`
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: var(--background-login);
`;

const StyledLogo = styled.img`
  width: 250px;
  height: 250px;
`;

const TitleStyled = styled.h1`
  font-size: var(--fs-md);
`;

const StyledLoginForm = styled.div`
  background-color: var(--form-color);

  border-radius: var(--radius-sm);

  display: flex;
  flex-direction: column;
  gap: 10px;

  margin-top: 20px;
  padding: 30px 50px;
`;

const FieldInput = styled.input`
  ${CommonInput}

  width: 300px;
`

export {
  LoginLayout,
  StyledLogo,
  TitleStyled,
  StyledLoginForm,
  FieldInput
};
