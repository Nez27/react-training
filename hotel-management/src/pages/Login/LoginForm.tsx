import { useForm } from 'react-hook-form';

// Components
import Form from '@src/components/Form';

// Styled
import { FieldInput, StyledLoginForm } from './styled';
import Button from '@src/commons/styles/Button';

// Constants
import {
  INVALID_EMAIL,
  REQUIRED_FIELD_ERROR,
} from '@src/constants/formValidateMessage';

// Types
import { ILogin } from '@src/types/common';

// Hooks
import { useLogin } from '@src/hooks/authentication/useLogin';

// Helpers
import { isValidRegex } from '@src/helpers/validators';

// Constants
import { REGEX } from '@src/constants/commons';

const LoginForm = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<ILogin>({
    defaultValues: {
      email: 'admin@nezumi.site',
      password: 'Phanhuuloi2@@1',
    }
  });
  const { login, isPending } = useLogin();

  const onSubmit = (data: ILogin) => {
    login(data);
  };

  return (
    <StyledLoginForm>
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Row
          label="Email:"
          direction="vertical"
          error={errors.email?.message}
        >
          <FieldInput
            type="text"
            id="email"
            {...register('email', {
              required: REQUIRED_FIELD_ERROR,
              validate: (value) =>
                isValidRegex(REGEX.EMAIL, value) || INVALID_EMAIL,
            })}
          />
        </Form.Row>
        <Form.Row
          label="Password:"
          direction="vertical"
          error={errors.password?.message}
        >
          <FieldInput
            type="password"
            id="password"
            {...register('password', {
              required: REQUIRED_FIELD_ERROR,
            })}
          />
        </Form.Row>

        <Button type="submit" disabled={isPending}>
          Login
        </Button>
      </Form>
    </StyledLoginForm>
  );
};

export default LoginForm;
