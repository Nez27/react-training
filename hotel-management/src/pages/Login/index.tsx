import { Navigate } from 'react-router-dom';

// Styled
import { LoginLayout, StyledLogo, TitleStyled } from './styled';

// Assets
import logo from '@assets/images/logo.png';

// Components
import LoginForm from './LoginForm';

// Hooks
import { useAccount } from '@hook/authentication/useAccount';

const Login = () => {
  const { account } = useAccount();

  if (account) {
    return <Navigate to={'/booking'} />;
  }

  return (
    <LoginLayout>
      <StyledLogo src={logo} />
      <TitleStyled>Log In</TitleStyled>

      <LoginForm />
    </LoginLayout>
  );
};

export default Login;
