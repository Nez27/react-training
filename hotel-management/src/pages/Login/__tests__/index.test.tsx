import { render } from '@testing-library/react';
import Login from '..';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router-dom';

describe('Login', () => {
  test('Should render correctly', () => {
    const queryClient = new QueryClient();

    const wrapper = render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </QueryClientProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
