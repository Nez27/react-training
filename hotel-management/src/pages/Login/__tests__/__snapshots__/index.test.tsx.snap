// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-bdnyFh fWdqRC"
      >
        <img
          alt="Logo Web"
          class="sc-gtsqUy cUcA-DD"
          src="./assets/images/logo.png"
        />
        <h1
          class="sc-dlniIP bJwJBk"
        >
          Log In
        </h1>
        <div
          class="sc-hKFymg htLQlQ"
        >
          <form
            class="sc-gKAaef fefJbB"
          >
            <div
              class="sc-fujznN fRJTpo"
              direction="vertical"
            >
              <label
                class="sc-pNWRh gSLqll"
              >
                Email:
              </label>
              <div>
                <input
                  class="sc-eCAqax gsyAlx"
                  id="email"
                  name="email"
                  type="text"
                />
              </div>
            </div>
            <div
              class="sc-fujznN fRJTpo"
              direction="vertical"
            >
              <label
                class="sc-pNWRh gSLqll"
              >
                Password:
              </label>
              <div>
                <input
                  class="sc-eCAqax gsyAlx"
                  id="password"
                  name="password"
                  type="password"
                />
              </div>
            </div>
            <button
              class="sc-jSFipO fQJaue"
              type="submit"
              variations="primary"
            >
              Login
            </button>
          </form>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-bdnyFh fWdqRC"
    >
      <img
        alt="Logo Web"
        class="sc-gtsqUy cUcA-DD"
        src="./assets/images/logo.png"
      />
      <h1
        class="sc-dlniIP bJwJBk"
      >
        Log In
      </h1>
      <div
        class="sc-hKFymg htLQlQ"
      >
        <form
          class="sc-gKAaef fefJbB"
        >
          <div
            class="sc-fujznN fRJTpo"
            direction="vertical"
          >
            <label
              class="sc-pNWRh gSLqll"
            >
              Email:
            </label>
            <div>
              <input
                class="sc-eCAqax gsyAlx"
                id="email"
                name="email"
                type="text"
              />
            </div>
          </div>
          <div
            class="sc-fujznN fRJTpo"
            direction="vertical"
          >
            <label
              class="sc-pNWRh gSLqll"
            >
              Password:
            </label>
            <div>
              <input
                class="sc-eCAqax gsyAlx"
                id="password"
                name="password"
                type="password"
              />
            </div>
          </div>
          <button
            class="sc-jSFipO fQJaue"
            type="submit"
            variations="primary"
          >
            Login
          </button>
        </form>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
